### PR TITLE
use constraints for `sload` & `sstore` gas

### PR DESCRIPTION
**Pull request by HAOYUatHZ**
close https://github.com/appliedzkp/zkevm-specs/issues/123

### CODE REVIEW

File: src/zkevm_specs/evm/execution/storage.py [hunk 1]
```diff
@@ -70,36 +70,63 @@ def sstore(instruction: Instruction):
...
+        instruction.select(
+            instruction.is_equal(original_value, value_prev),
+            instruction.select(
+                instruction.is_zero(original_value),
+                SSTORE_SET_GAS,
+                SSTORE_RESET_GAS,
+            ),
+            SLOAD_GAS,
+        ),
+    )
```
Comment (HAOYUatHZ) [inline src/zkevm_specs/evm/execution/storage.py:128]: in fact, this part can be simplified as 
```
warm_case_gas = instruction.select(
    instruction.is_equal(value_prev, value) | (~ instruction.is_equal(original_value, value_prev)),
    SLOAD_GAS,
    instruction.select(
        instruction.is_zero(original_value),
        SSTORE_SET_GAS,
        SSTORE_RESET_GAS,
    ),
)
```

but the simplified version doesn't reflect the logic in https://eips.ethereum.org/EIPS/eip-2200 directly, so I don't know which version I should keep.

Would like to apply it if we think the simplified version is better

Comment (ChihChengLiang) [reply src/zkevm_specs/evm/execution/storage.py:128]: Does the ` | ` in the simplified version work in the circuit?

**Comment by ChihChengLiang** [comment]
@HAOYUatHZ Can we do a rebase before the review?

File: src/zkevm_specs/evm/execution/storage.py (hunk 1 above)
Comment (HAOYUatHZ) [reply src/zkevm_specs/evm/execution/storage.py:128]: The logic is the same. It basically merges these two parts
1) L117~118 
2) L121~126 
together

**Comment by HAOYUatHZ** [comment]
> @HAOYUatHZ Can we do a rebase before the review?

sure.

File: src/zkevm_specs/evm/execution/storage.py (hunk 1 above)
Comment (ChihChengLiang) [reply src/zkevm_specs/evm/execution/storage.py:128]: I see. The simplified version looks better to me. Will hold the review after you add it.

**Review by ChihChengLiang — APPROVED** [review]
LGTM

**Review by CPerezz — APPROVED** [review]
LGTM

**Pull request merged**